torch_np/_funcs_impl.py

```diff
@@ -894,6 +894,29 @@ def take_along_axis(arr: ArrayLike, indices: ArrayLike, axis):
     return torch.take_along_dim(arr, indices, axis)
 
 
+def put(
+    a: NDArray,
+    ind: ArrayLike,
+    v: ArrayLike,
+    mode: NotImplementedType = "raise",
+):
+    v = v.type(a.dtype)
+    # If ind is larger than v, broadcast v to the would-be resulting shape. Any
+    # unnecessary trailing elements are then trimmed.
+    if ind.numel() > v.numel():
+        ratio = (ind.numel() + v.numel() - 1) // v.numel()
+        sizes = [ratio]
+        sizes.extend([1 for _ in range(v.dim() - 1)])
+        v = v.repeat(*sizes)
```

Comment:
Nit. `[1] * (v.dim() - 1)`.

That being said, as discussed inhttps://github.com/Quansight-Labs/numpy_pytorch_interop/pull/116#discussion_r1172316059, what you probably want is an unsqueeze(0) + expand, as you don't want to copy the data if you can avoid it.

Reply:
Ah thanks I see, pushed a change to this effect.